src/sbt-test/sbt-1.4/relax-scalacOptions/test

```diff
@@ -0,0 +1,48 @@
+# TODO: confirm that we shouldn't rewire scalafix/compile in the right task scope to see compileInputsSettings?
+# TODO: earlycompilation/pipelining support
+
+# when the build does not contain any flag relaxed on scalafix invocation
+# check that regular compilation benefits froma prior direct scalafix invocation ...
+-> scalafix --check RemoveUnused
+-> checkLastCompilationCached
+> compile
+> checkLastCompilationCached
+
+# ... and indirect scalafix invocation
+> clean
+> test:scalafix --check RemoveUnused
+-> checkLastCompilationCached
+> compile
+> checkLastCompilationCached
+
+# we now have 2 fatal warnings preventing compilation
+> set scalacOptions ++= Seq("-Xfatal-warnings")
+-> compile
+
+# ensure that a semantic rule can be ran despite warnings, to fix one of the warning
+> scalafix RemoveUnused
+
+# confirm that the rule was applied and that triggered compilation is cached once the updated file recompiled
+> scalafix --check RemoveUnused
+-> checkLastCompilationCached
+> scalafix --check RemoveUnused
+> checkLastCompilationCached
```

Comment:
successive `scalafix` invocations on a cold environment (where no `compile` was run successfully) now benefit from incremental compilation, no matter which flags are used in the build